pkg/controller/atlasdeployment/advanced_deployment.go

```diff
@@ -77,12 +81,94 @@ func (r *AtlasDeploymentReconciler) ensureAdvancedDeploymentState(ctx *workflow.
 	}
 }
 
+func handleSearchNodes(ctx *workflow.Context, deployment *mdbv1.AtlasDeployment, projectID string) error {
```

Comment:
I think it makes sense to move this to its own file `searchnodes.go`

Reply:
agreed